### ISSUE TITLE
correct regex in /etc/fail2ban/filter.d/sshd.conf

### ISSUE THREAD

**Issue by deutrino** [turnkeylinux/tracker#1473]
An existing set of regex patterns in /etc/fail2ban/filter.d/sshd.conf is missing one to match the actual output of the ssh daemon installed in Turnkey 16.

To that file, to the 'mdre-ddos' variable, add the following:

            ^Did not receive identification string from <HOST>%(__on_port_opt)s$

**Comment by deutrino** [comment]
Now I suppose I should figure out how to make a pull request vs core??

**Comment by JedMeister** [comment]
Interesting, our fail2ban config is actually inherited from Debian! So it could be argued that it's actually an upstream bug?! Having said that, perhaps there's something I missed in the docs when I included it?!

Regardless, there is already both a [fail2ban-fix overlay](https://github.com/turnkeylinux/common/tree/master/overlays/turnkey.d/fail2ban-fix) and [fail2ban-fix conf script](https://github.com/turnkeylinux/common/blob/master/conf/turnkey.d/fail2ban-fixes). Both of these are applied to all appliances. The former is overlaid relative to `/` - i.e. a file in `common/overlays/turnkey.d/fail2ban-fix/etc/somefile` would end up as `/etc/somefile`. The conf script is run inside `/`. The common overlays are all done first (in alphanumeric order), then the common conf scripts (again in alphanumeric order). Hope that helps...

**Comment by OnGle** [comment]
Any further news on this?

**Comment by JedMeister** [comment]
@OnGle - nope, but we should revisit it for v17.0 IMO. Once the Core & TKLDev rc is released, I'll aim to have a look (as such will self assign).

**Comment by JedMeister** [comment]
@deutrino - I was just looking into this for v17.0 and I note that the first entry of `mdre-ddos` is a line:
```
^Did not receive identification string from <HOST>
```
I have had a quick look at the regex used by fail2ban with the hope that I might be able to confirm (or not) that this regex meets your requirements. However, I can't find a quick and easy guide to how the fail2ban regex works...

Could you confirm (or not) that we still need this line that you suggest (please provide a little info too).

FWIW, the default v16.x fail2ban ssh conf includes:
```
^Did not receive identification string from <HOST>%(__suff)s$
```
which I note is closer, but still not the same as yours...

**Comment by JedMeister** [comment]
@deutrino - any more input into this? I'll likely be pushing out v17.0 Core and TKLDev (at least in ISO and EC2) within the next few days, so we've possibly missed the boat for inclusion in those, but I'd be happy to add this (or similar- as/if needed) as soon as we can have confirmation that it's required.

**Comment by JedMeister** [comment]
I'm going to close this as AFAIK this particular issue was resolved in v17.0.
